core/src/main/java/com/scalar/db/api/AuthAdmin.java

```diff
@@ -404,48 +404,68 @@ interface User {
     boolean isSuperuser();
   }
 
-  /** Represents a role. */
+  /** Represents a role, including its granted member roles. */
   interface Role {
+    /**
+     * Returns the role name.
+     *
+     * @return the role name
+     */
     String getName();
-  }
 
-  /** Represents a role with its hierarchy information. */
-  interface RoleDetail {
-    Role getRole();
-
-    List<RoleHierarchy> getRoleHierarchies();
+    /**
+     * Returns the member roles granted to the role.
+     *
+     * @return the member roles granted to the role
+     */
+    List<MemberRole> getMemberRoles();
   }
 
   /**
-   * Represents a role detail for a specific user, including whether the user has admin option for
+   * Represents a role granted to a specific user, including whether the user has admin option for
    * this role.
    */
-  interface UserRoleDetail extends RoleDetail {
+  interface RoleForUser extends Role {
     /**
      * Returns whether the user has admin option for this role. This is distinct from the admin
      * option in role hierarchies, which applies to role-to-role grants.
+     *
+     * @return whether the user has admin option for this role
      */
     boolean hasAdminOptionOnUser();
   }
 
   /** Represents a user-role assignment. */
-  interface UserRole {
-    String getUsername();
-
-    String getRoleName();
+  interface GranteeUser {
+    /**
+     * Returns the username.
+     *
+     * @return the username
+     */
+    String getName();
 
+    /**
+     * Returns whether admin option is granted for this assignment.
+     *
+     * @return whether admin option is granted for this assignment
+     */
     boolean hasAdminOption();
   }
 
   /** Represents a role hierarchy (role-to-role assignment). */
-  interface RoleHierarchy {
-    /** Returns the role name. */
-    String getRoleName();
-
-    /** Returns the member role name granted to the role. */
-    String getMemberRoleName();
+  interface MemberRole {
```

Comment:
This looks more like a reference than the actual role, so does it make sense to name it `RoleReference` or `RoleRef`?